### PR TITLE
Completed HW 34-35

### PR DESCRIPTION
**Pull request by RomanMinevich**

### CODE REVIEW

File: src/main/java/mate/academy/spring/MainApp.java [hunk 1]
```diff
@@ -1,79 +1,16 @@
...
-import mate.academy.spring.entity.Author;
-import mate.academy.spring.entity.Book;
-import mate.academy.spring.entity.User;
-import mate.academy.spring.service.AuthorService;
-import mate.academy.spring.service.BookService;
-import mate.academy.spring.service.RentService;
-import mate.academy.spring.service.UserService;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
 public class MainApp {
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/MainApp.java]: @RomanMinevich feel free to remove it

File: src/main/java/mate/academy/spring/annotations/Valid.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Retention(RUNTIME)
+@Constraint(validatedBy = EmailValidator.class)
+@Documented
+public @interface Valid {
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/annotations/Valid.java]:
```suggestion
public @interface ValidEmail {
```

File: src/main/java/mate/academy/spring/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+public class CustomUserDetailsService implements UserDetailsService {
+    @Autowired
+    private UserDao userDao;
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+
+        User user = userDao.findByUsername(username);
+        if (user == null) {
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/security/CustomUserDetailsService.java]: @RomanMinevich Can we return optional _ findByUsername_?

File: src/main/java/mate/academy/spring/security/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            throw new UsernameNotFoundException("User not found");
+        }
+        return new org.springframework.security.core.userdetails.User(
+                user.getEmail(),
+                user.getPassword(),
+                getAuthorities(getRoleNames(user.getRoles())));
+    }
+
+    private static List<String> getRoleNames(List roles) {
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/security/CustomUserDetailsService.java]: @RomanMinevich Can we use a parametrized List?

File: src/main/java/mate/academy/spring/security/CustomUserDetailsService.java (hunk 3 above)
Comment (RomanMinevich) [reply src/main/java/mate/academy/spring/security/CustomUserDetailsService.java]: Done

File: src/main/java/mate/academy/spring/annotations/Valid.java (hunk 2 above)
Comment (RomanMinevich) [reply src/main/java/mate/academy/spring/annotations/Valid.java]: Done

File: src/main/java/mate/academy/spring/MainApp.java (hunk 1 above)
Comment (RomanMinevich) [reply src/main/java/mate/academy/spring/MainApp.java]: Removed

File: src/main/java/mate/academy/spring/security/CustomUserDetailsService.java (hunk 4 above)
Comment (boroda4436) [reply src/main/java/mate/academy/spring/security/CustomUserDetailsService.java]: @RomanMinevich please post this question into slack group channel?

Comment (RomanMinevich) [reply src/main/java/mate/academy/spring/security/CustomUserDetailsService.java]: May I ask for help in retrospective first?

Comment (RomanMinevich) [reply src/main/java/mate/academy/spring/security/CustomUserDetailsService.java]: Fixed

File: src/main/java/mate/academy/spring/MainApp.java (hunk 1 above)
Comment (Sofasmile) [reply src/main/java/mate/academy/spring/MainApp.java]: did you remove this class?

**Pull request merged**